dualscreeninfo/android/src/main/kotlin/com/microsoft/reactnativedualscreen/dualscreen/DualScreenInfo.kt

```diff
@@ -0,0 +1,138 @@
+package com.microsoft.reactnativedualscreen.dualscreen
+
+import android.content.Context
+import android.graphics.Rect
+import android.view.View
+import android.view.WindowManager
+import com.facebook.react.bridge.*
+import com.facebook.react.bridge.Arguments.createMap
+import com.facebook.react.modules.core.DeviceEventManagerModule.RCTDeviceEventEmitter
+import com.microsoft.device.display.DisplayMask
+
+const val HINGE_WIDTH_KEY = "hingeWidth"
+const val IS_DUALSCREEN_DEVICE_KEY = "isDualScreenDevice"
+const val FEATURE_NAME = "com.microsoft.device.display.displaymask"
+
+class DualScreenInfo constructor(context: ReactApplicationContext) : ReactContextBaseJavaModule(context) {
+    private val isDualScreenDevice: Boolean
+        get() =  reactApplicationContext.packageManager.hasSystemFeature(FEATURE_NAME)
+	private val mDisplayMask: DisplayMask
+		get() = DisplayMask.fromResourcesRectApproximation(currentActivity)
+	private val rotation: Int
+		get() {
+			val wm = currentActivity?.getSystemService(Context.WINDOW_SERVICE) as WindowManager?
+			return wm?.defaultDisplay?.rotation ?: 0
+		}
+	private val hinge: Rect
+		get() {
+			// Hinge's coordinates of its 4 edges in different mode
+			// Double Landscape Rect(0, 1350 - 1800, 1434)
+			// Double Portrait  Rect(1350, 0 - 1434, 1800)
+			return mDisplayMask.getBoundingRectsForRotation(rotation)[0]
+		}
+
+	override fun getName() = "DualScreenInfo"
+
+	override fun initialize() {
+		super.initialize()
+		createSubscription()
+	}
+
+	override fun getConstants(): Map<String, Any>? {
+        val constants: MutableMap<String, Any> = HashMap()
+    	constants[HINGE_WIDTH_KEY] = 34
```

Comment:
>34 [](start = 34, length = 2)

Why 34?

Reply:
34dp is official size of Hinge on Duo device - now its only used for ViewPager's current implementation

Reply:
Sorry I'm not following...  Is this width of the gap between screens?  If so why can't you compute from the displayMask API?  We shouldn't have to hard-code any such values.

---
In reply to: [382724045](https://github.com/microsoft/react-native-dualscreen/pull/2#discussion_r382724045) [](ancestors = 382724045)

Reply:
Agreed that we shouldn't hardcode numbers and instead should query the [Duo API](https://docs.microsoft.com/en-us/dual-screen/android/api-reference/display-mask?tabs=java). Other dual-screen devices will exist in the future, so 34dp is not necessarily a safe assumption to make for all of them.

Reply:
@kmelmon be default app always starting on single screen, so hinge is always 0 and I can't detect it programmatically on start. And since react-native-viewpager supports only pre-defined value, I think we can keep it until it will be fixed